deep_qa/training/text_trainer.py

```diff
@@ -305,7 +306,13 @@ def _embed_input(self, input_layer: Layer, embedding_name: str="embedding"):
         uses both words and characters, we need to run the character encoder and concatenate the
         result with a word embedding).
         """
-        return self.tokenizer.embed_input(input_layer, self.__get_embedded_input, self, embedding_name)
+        with tensorflow.device("/cpu:0"):
```

Comment:
I think this is the wrong place for this.  If you're using a `word and characters` tokenizer, `tokenizer.embed_input` gets a word embedding and also does a character-level CNN, then concatenates them.  You only want the embedding lookups to happen on the CPU, not the CNN.  I also am thinking that MattP's pre-trained language model stuff will fit into DeepQA in the call to `self._embed_input`, so this could also get a lot more complex in some scenarios.  I think the right place to put the device scope is [here](https://github.com/allenai/deep_qa/blob/00d36306759cb1c232489f68844371fb727ce2c8/deep_qa/training/text_trainer.py#L602).